### PR TITLE
Yet more desktop tests

### PR DESCRIPTION
**Pull request by rainersigwald**
There are three in here and three remain unchecked in #799.

/giphy I want to believe

### CODE REVIEW

**Comment by rainersigwald** [comment]
Ok, I have no idea why the strong/weak reference test is failing. Happens consistently on my machine too, but there doesn't seem to be any code change from master . . .

**Comment by rainersigwald** [comment]
I (sorta) figured out the weak-reference test. For some reason, if we check for mono and early-return, an additional strong reference is detected and the GC doesn't clean up, breaking the weak references.

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build for CoreCLR please

**Comment by rainersigwald** [comment]
@cdmihai @jeffkl @AndyGerlicher Looks good to review now.

**Comment by rainersigwald** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR please

File: src/XMakeCommandLine/UnitTests/XMake_Tests.cs [hunk 1]
```diff
@@ -801,6 +801,26 @@ public void GetCommandLine()
             Assert.True(File.Exists(_pathToArbitraryBogusFile));
 
             bool successfulExit;
+            string output = RunnerUtilities.ExecMSBuild('"' + RunnerUtilities.PathToCurrentlyRunningMsBuildExe + '"', msbuildParameters, out successfulExit);
+            Assert.False(successfulExit);
+
+            Assert.Contains(RunnerUtilities.PathToCurrentlyRunningMsBuildExe + (NativeMethodsShared.IsWindows ? " /v:diag " : " -v:diag ") + _pathToArbitraryBogusFile, output, StringComparison.OrdinalIgnoreCase);
```
Comment (cdmihai) [inline src/XMakeCommandLine/UnitTests/XMake_Tests.cs]: I think these these are the other way around. `GetCommandLine` does not quote msbuild.exe, but`GetCommandLineQuotedExe` quotes it

**Comment by cdmihai** [comment]
@dotnet-bot test this please

File: src/XMakeCommandLine/UnitTests/XMake_Tests.cs (hunk 1 above)
Comment (rainersigwald) [inline src/XMakeCommandLine/UnitTests/XMake_Tests.cs]: Nice catch! Fixed in latest iteration.

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build for CoreCLR  please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build for CoreCLR please

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build for CoreCLR please

**Comment by rainersigwald** [comment]
Tests pass on OS X on my laptop. @jeffkl is investigating the CI failure. Plowing forward.

**Pull request merged**